src/arcaflow_plugin_sdk/test_schema.py

```diff
@@ -963,6 +963,8 @@ class TestData1:
                 default="", metadata={"id": "test-field", "name": "G"}
             )
             I: typing.Any = None
+            J: typing.List[typing.Dict[str, str]] \
+                = dataclasses.field(default_factory=list)
 
         schema.test_object_serialization(
             TestData1(A="Hello world!", B=5, C={}, D=[], H=3.14),
```

Comment:
Not gonna try a `J=[{}]` here?? (`I` is missing, too ... and a second serialization that doesn't omit the `Optional`s would be good as well.)

Reply:
LOL.  I had [the same question](https://github.com/arcalot/arcaflow-plugin-sdk-python/pull/141#discussion_r1834649718).  😁

Reply:
And of course it'd be grand to verify that `J=[{"a":1},{"b":2}]` worked correctly while it ought to complain about `J=[{1:"a"}]`. 😆

Reply:
Sure. I just pushed a chance that added that field.

Reply:
Cool. (And of course I meant `[str, str]` but my brain was thinking about Dustin's case!) 😁 

Optional and failure cases would be nice, too, and presumably bad `list[dict[...]]` cases *aren't* already tested, since it wouldn't have worked right. But as a quick fix this doesn't look bad.